fix: Allow views with `Nullable(Nothing)` columns to be materialized

rafaeelaudibert opened the pull request:
Cast `Nullable(Nothing)` columns to `Nullable(String)` before attempting to store them in Arrow since `Nullable(Nothing)` isn't supported. This doesn't have any impact because we only have `Nullable(Nothing)` if CH knows the column is NULL for every single value (something like `SELECT NULL AS column`)
    
This isn't a breaking change because these tables were failing to be materialized anyways

rafaeelaudibert committed:
feat: Add `accurateCastOrNull` CH function

We don't expose any `cast`-related to customers but I don't know why. This is very helpful in a lot of situations so let's make it available for everyone, happy to be convinced this is wrong

rafaeelaudibert committed:
fix: Allow views with `Nullable(Nothing)` columns to be materialized

Cast `Nullable(Nothing)` columns to `Nullable(String)` before attempting to store them in Arrow since `Nullable(Nothing)` isn't supported. This doesn't have any impact because we only have `Nullable(Nothing)` if CH knows the column is NULL for every single value (something like `SELECT NULL AS column`)

This isn't a breaking change because these tables were failing to be materialized anyways

rafaeelaudibert committed:
tests: Add tests for `Nullable(Nothing)` construct

github-actions[bot] committed:
Update query snapshots

rafaeelaudibert committed:
fix: fix tests

Gilbert09 reviewed (changes requested):
Just the one change on adding `accurateCastOrNull`

rafaeelaudibert committed:
refactor: Avoid using `accurateCastOrNull`

@Gilbert09 isn't open to exposing `accurateCastOrNull` for all users, so let's instead create a custom `toNullableString` to circumvent the problems we have.

For context, we can't use `toString` directly because `toString(x)` where `x` is `Nullable(Nothing)` (i.e. a constant `NULL`) still returns `Nullable(Nothing)` as type.

rafaeelaudibert commented on posthog/hogql/functions/clickhouse/conversions.py:
```diff
@@ -88,6 +88,8 @@
     "cityHash64": HogQLFunctionMeta("cityHash64", 1, 1),
     "UUIDv7ToDateTime": HogQLFunctionMeta("UUIDv7ToDateTime", 1, 1, tz_aware=True),
+    # Expose the raw function, this is slightly safer than `cast` but should solve most problems
+    "accurateCastOrNull": HogQLFunctionMeta("accurateCastOrNull", 2, 2),
```
@Gilbert09 Pushed 909103f3a1f63640243d82de574e7e48c82170d9 with a possible fix. We can't use `toString` here, you can see the commit message for the rationale

github-actions[bot] committed:
Update query snapshots
